contributing/contribute-code.md

```diff
@@ -0,0 +1,209 @@
+---
+title:  "Contributing Code"
+---
+
+Apache Flink is maintained, improved, and extended by code contributions of volunteers. We welcome contributions to Flink, but due to the size of the project and to preserve the high quality of the code base, we follow a contribution process that is explained in this document.
+
+**Please feel free to ask questions at any time.** Either send a mail to the [dev mailing list]( {{ site.base }}/community.html#mailing-lists ) or comment on the Jira issue you are working on.
+
+**IMPORTANT**: Please read this document carefully before starting to work on a code contribution. Follow the process and guidelines explained below. Otherwise, your pull request might not be accepted or might require substantial rework. In particular, before opening a pull request that implements a **new feature**, you need to open a Jira ticket and reach consensus with the community on whether this feature is needed.
+
+
+
+{% toc %}
+
+## Code Contribution Process
```

Comment:
Regarding the overall process: does it really make sense to indicate accepted Jiras by assigning somebody? 
What about cases when there is nobody to immediately work on an issue or if the assignee wants to be unassigned later? Will the issue lose it's consensus?

IMO, it would make more sense to add a tag "Accepted" or "Consensus Reached". IIRC, we tried that like 3 years ago.

Reply:
The problem with using tags is that everybody can modify them.
The good thing about assigning somebody is that this is an action that can be controlled as a permission.

In cases where we unassign somebody, it will still be pretty clear from the discussion that there is consensus. 
I would propose to look into solutions for this if it is really a practical problem.